students/rachel_s/session07/html_render.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+from string import Template
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+
+    tag = "html"
+
+    indent = '    '
+
+    def __init__(self, content=None, **kwargs):
+        self.contents = []
+        self.args = ''
+        if content:
+            self.contents.append(content)
+        if kwargs:
+            self.args = kwargs
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+    def _indenter(self, cur_ind):
+        output = ''
+        for i in range(cur_ind):
+            output += self.indent
+        return output
+
+    def render(self, out_file, cur_ind=0):
+        out_file.write(self._indenter(cur_ind))
+        out_file.write(self._open_tag())
+        out_file.write('\n')
+        for item in self.contents:
+            try:
+                item.render(out_file, cur_ind=cur_ind+1)
+            except AttributeError:
+                out_file.write(self._indenter(cur_ind+1))
+                out_file.write(item)
+            out_file.write("\n")
+        out_file.write(self._indenter(cur_ind))
+        out_file.write(self._close_tag())
+
+    def _unpack_kwargs(self):
+        output = ''
+        for key, value in self.args.items():
+            output += ' {par}=\"{val}\"'.format(par=key, 
+            val=str(value))
+        return output
```

Comment:
Minor: Consider using a list comprehension for things like this, like `return ' '.join('{par}="{val}"'.format(par=key, val=str(value)) for key, value in self.args.items())`. Of course you'll still need to handle the leading blank space in this case, when self.args is not empty.